src/BaseBlockTracker.ts

```diff
@@ -56,16 +63,38 @@ export abstract class BaseBlockTracker extends SafeEventEmitter {
   }
 
   async getLatestBlock(): Promise<string> {
-    // return if available
     if (this._currentBlock) {
       return this._currentBlock;
     }
-    // wait for a new latest block
-    const latestBlock: string = await new Promise((resolve) =>
-      this.once('latest', resolve),
-    );
-    // return newly set current block
-    return latestBlock;
+
+    const promiseForLatestBlock = new Promise<string>((resolve) => {
+      // eslint-disable-next-line prefer-const
+      let errorHandler: (error: unknown) => void;
+      // eslint-disable-next-line prefer-const
+      let successHandler: (blockNumber: string) => void;
+      // eslint-disable-next-line prefer-const
+      errorHandler = (error: unknown) => {
+        console.error(error);
```

Comment:
It's not entirely clear why we need to capture the error here when we hadn't done so previously. The error we get might be related to another call.